lib/train/transports/ssh_connection.rb

```diff
@@ -29,8 +29,8 @@ class Train::Transports::SSH
   #
   # @author Fletcher Nichol <fnichol@nichol.ca>
   class Connection < BaseConnection # rubocop:disable Metrics/ClassLength
-    attr_reader :hostname
-    attr_reader :transport_options
+    attr_reader   :hostname
+    attr_accessor :transport_options
```

Comment:
Could you help me understand why this change is needed? I don't see anywhere in the PR what the transport_option ivar is directly changed, and if you need to change one of its elements you can do so without making it read-write.

Reply:
Here's the context where the goal is to change the pty option to true later in the process:

https://github.com/chef/chef/pull/10410/files#r492238172

I had the same question, but didn't know enough about train to suggest anything better.

Reply:
Does this work OK with the ssh code [here](https://github.com/inspec/train/blob/master/lib/train/transports/ssh.rb#L81) that looks at the options to determine if the connection can be re-used or needs restarted?

Reply:
@james-stocks As we memoinzed the connection object at chef core side https://github.com/chef/chef/blob/0a29892953b07ba1fded119f5aef364e14c7dd07/lib/chef/knife/bootstrap/train_connector.rb#L51-L59, won't be a problem here, but not sure about other impacts.